File: size_test/src/golang/complex/complex.go
package main

import (
	"fmt"
	"sort"
)

type ComplexStruct struct {
	Data    []string
	Mapping map[int]string
}

func (c *ComplexStruct) AddData(item string) {
	c.Data = append(c.Data, item)
	if c.Mapping == nil {
		c.Mapping = make(map[int]string)
	}
	c.Mapping[len(c.Data)] = item
}

func (c *ComplexStruct) ProcessData() {
	sort.Strings(c.Data)
	for _, item := range c.Data {
		fmt.Printf("Processed: %s\n", item)
	}
}

func main() {
	obj := &ComplexStruct{}
	obj.AddData("Hello")
	obj.AddData("World")
	obj.AddData("From")
	obj.AddData("Complex")
	obj.AddData("Go")
	obj.ProcessData()
}

File: size_test/src/cpp/complex.cpp
#include <iostream>
#include <vector>
#include <string>
#include <map>
#include <algorithm>

class ComplexClass {
private:
    std::vector<std::string> data;
    std::map<int, std::string> mapping;
    
public:
    void addData(const std::string& item) {
        data.push_back(item);
        mapping[data.size()] = item;
    }
    
    void processData() {
        std::sort(data.begin(), data.end());
        for (const auto& item : data) {
            std::cout << "Processed: " << item << std::endl;
        }
    }
};

int main() {
    ComplexClass obj;
    obj.addData("Hello");
    obj.addData("World");
    obj.addData("From");
    obj.addData("Complex");
    obj.addData("C++");
    obj.processData();
    return 0;
}
